Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -369,17 +369,14 @@ void Document::visit_edges(Cell::Visitor& visitor)
 }
 
 // https://w3c.github.io/selection-api/#dom-document-getselection
-JS::GCPtr<Selection::Selection> Document::get_selection()
+WebIDL::ExceptionOr<JS::GCPtr<Selection::Selection>> Document::get_selection()
 {
     // The method must return the selection associated with this if this has an associated browsing context,
     // and it must return null otherwise.
-    if (!browsing_context()) {
+    if (!browsing_context())
         return nullptr;
-    }
-
-    if (!m_selection) {
-        m_selection = Selection::Selection::create(realm(), *this).release_value_but_fixme_should_propagate_errors();
-    }
+    if (!m_selection)
+        m_selection = MUST_OR_THROW_OOM(heap().allocate<Selection::Selection>(realm(), realm(), *this));
```

Comment:
Since we use the DOM selection for the painted selection as well now, this probably doesn't need to be lazily allocated anymore.
We could allocate `m_selection` in `initialize()` and then this API doesn't have to be throwsy. Not necessary to fix in this PR, although it would make this commit a bit simpler :)